local-volume/onetimecontroller/controller.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func updatePVAlphaAnn(client *kubernetes.Clientset, pv *v1.PersistentVolume) error {
+	if len(pv.Annotations) > 0 && pv.Annotations[v1.AlphaStorageNodeAffinityAnnotation] != "" {
+		var affinity v1.NodeAffinity
+		err := json.Unmarshal([]byte(pv.Annotations[v1.AlphaStorageNodeAffinityAnnotation]), &affinity)
+		if err != nil {
+			return err
+		}
+		pvClone := pv.DeepCopy()
+		delete(pvClone.Annotations, v1.AlphaStorageNodeAffinityAnnotation)
+		pvClone.Spec.NodeAffinity = &v1.VolumeNodeAffinity{
+			Required: affinity.RequiredDuringSchedulingIgnoredDuringExecution,
+		}
+
+		_, err = client.CoreV1().PersistentVolumes().Update(pvClone)
```

Comment:
When you attempt to update a large set of PVs, it is quite likely that one of the PVs might have been updated (perhaps released or bound) between the time you listed it out and the time you iterate to each PV and actually update the PV. This will give rise to "409 conflict error" when you try to update it using a clone of its outdated version. You can detect this error using errors.IsConflict(err) method. You can resolve this issue by GETing the latest resource for the PV and retrying the update.

Reply:
One more thing to consider when updating a large about of objects.  You may potentially hit API quota.   In that case, we may need to retry the update a few times.

Reply:
thanks, done